iotdb-client/cli/src/main/java/org/apache/iotdb/tool/ImportTsFileRemotely.java

```diff
@@ -0,0 +1,287 @@
+      params.put(
+          PipeTransferHandshakeConstant.HANDSHAKE_KEY_CONVERT_ON_TYPE_MISMATCH,
+          Boolean.toString(true));
+      params.put(PipeTransferHandshakeConstant.HANDSHAKE_KEY_LOAD_TSFILE_STRATEGY, loadStrategy);
+
+      TPipeTransferResp resp =
+          client.pipeTransfer(PipeTransferDataNodeHandshakeV2Req.toTPipeTransferReq(params));
+
+      if (resp.getStatus().getCode() == TSStatusCode.PIPE_TYPE_ERROR.getStatusCode()) {
+        LOGGER.info(
+            "Handshake error with target server ip: {}, port: {}, because: {}. "
+                + "Retry to handshake by PipeTransferHandshakeV1Req.",
+            client.getIpAddress(),
+            client.getPort(),
+            resp.getStatus());
+        resp =
+            client.pipeTransfer(
+                PipeTransferDataNodeHandshakeV1Req.toTPipeTransferReq(
+                    CommonDescriptor.getInstance().getConfig().getTimestampPrecision()));
+      }
+
+      if (resp.getStatus().getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+        LOGGER.warn(
+            "Handshake error with target server ip: {}, port: {}, because: {}.",
+            client.getIpAddress(),
+            client.getPort(),
+            resp.getStatus());
+      } else {
+        client.setTimeout(CONNECTION_TIMEOUT_MS.get());
+        LOGGER.info(
+            "Handshake success. Target server ip: {}, port: {}",
+            client.getIpAddress(),
+            client.getPort());
+      }
+    } catch (final Exception e) {
+      ioTPrinter.println(
+          String.format(
+              "Handshake error with target server ip: %s, port: %s, because: %s.",
+              client.getIpAddress(), client.getPort(), e.getMessage()));
+    }
+  }
+
+  public void doTransfer(final File tsFile, final File modFile) throws PipeException, IOException {
+    final TPipeTransferResp resp;
+
+    if (Objects.nonNull(modFile)) {
+      transferFilePieces(modFile, true);
+      transferFilePieces(tsFile, true);
+
+      try {
+        final TPipeTransferReq req =
+            PipeTransferTsFileSealWithModReq.toTPipeTransferReq(
+                modFile.getName(), modFile.length(), tsFile.getName(), tsFile.length());
+
+        resp = client.pipeTransfer(req);
+      } catch (final Exception e) {
+        throw new PipeConnectionException(
+            String.format("Network error when seal file %s, because %s.", tsFile, e.getMessage()),
+            e);
+      }
+    } else {
+      transferFilePieces(tsFile, false);
+
+      try {
+        final TPipeTransferReq req =
+            PipeTransferTsFileSealReq.toTPipeTransferReq(tsFile.getName(), tsFile.length());
+
+        resp = client.pipeTransfer(req);
+      } catch (final Exception e) {
+        throw new PipeConnectionException(
+            String.format("Network error when seal file %s, because %s.", tsFile, e.getMessage()),
+            e);
+      }
+    }
```

Comment:
```suggestion
      final boolean modFileIsNull = Objects.nonNull(modFile);
      if(modFileIsNull){
        transferFilePieces(modFile, true);
      transferFilePieces(tsFile, true);
      }else{
        transferFilePieces(tsFile, false);
      }
      
      try {
      final TPipeTransferReq req = modFileIsNull ? 
            PipeTransferTsFileSealWithModReq.toTPipeTransferReq(modFile.getName(), modFile.length(), tsFile.getName(), tsFile.length())
                :PipeTransferTsFileSealReq.toTPipeTransferReq(tsFile.getName(), tsFile.length());
        resp = client.pipeTransfer(req);
      } catch (final Exception e) {
        throw new PipeConnectionException(
            String.format("Network error when seal file %s, because %s.", tsFile, e.getMessage()),
            e);
      }
```

Reply:
Already edited, you wrote it wrong. @luoluoyuyu ❤️